adapter/pokemon_adapter.go

```diff
@@ -0,0 +1,62 @@
+package adapter
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/nibble-4bits/ondemand-go-bootcamp/entity"
+)
+
+type pokemonAdapter struct {
+	dataSource DataSource
+	pokemons   []entity.Pokemon
+}
+
+func NewPokemonAdapter(ds DataSource) *pokemonAdapter {
```

Comment:
Be careful, you are returning an unexported field. This might be annoying to use.